### ISSUE TITLE
Property on metaclass should take precedence over property on class

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5686]
Pyright evaluates the wrong type in the case where a property of the same name is defined on both the metaclass and the class.

```python
class MyMeta(type):
    @property
    def attr1(cls) -> int:
        return 1

    @property
    def attr3(cls) -> int:
        return 3


class A(metaclass=MyMeta):
    @property
    def attr2(self) -> int:
        return 2

    @property
    def attr3(self) -> int:
        return 3


reveal_type(A.attr1, expected_text="int")
reveal_type(A().attr2, expected_text="int")
reveal_type(A.attr2, expected_text="property")
reveal_type(A().attr3, expected_text="int")
reveal_type(A.attr3, expected_text="int")  # This is incorrect
```

**Comment by erictraut** [comment]
This is related to a bug in mypy's issue tracker: https://github.com/python/mypy/issues/7945.

**Comment by erictraut** [comment]
This is included in pyright 1.1.324, which I just published. It will also be included in a future release of pylance.
